File: scripts/NativeCast.m
// File ObjC che registra esplicitamente NativeCastPlugin con Capacitor.
//
// L'auto-discovery di Capacitor 8 tramite CAPBridgedPlugin NON funziona per
// plugin Swift in-app aggiunti dinamicamente al progetto Xcode da pipeline
// CI (verificato sperimentalmente: la classe risulta compilata e caricata
// nella ObjC runtime, ma Capacitor non la trova → "plugin is not implemented
// on ios"). La macro CAP_PLUGIN genera il codice di registrazione che
// Capacitor cerca al boot — è lo stesso approccio usato da tutti i plugin
// ufficiali (@capacitor/camera, @capacitor/geolocation, ecc.).
//
// IMPORTANTE: il primo argomento (NativeCastPlugin) DEVE corrispondere
// all'@objc(NativeCastPlugin) della classe Swift. Il secondo argomento
// ("NativeCast") DEVE corrispondere al jsName e a registerPlugin("NativeCast")
// lato JS.

#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(NativeCastPlugin, "NativeCast",
    CAP_PLUGIN_METHOD(initialize,        CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(showPickerAndLoad, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(getDiagnostics,    CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(endSession,        CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(getState,          CAPPluginReturnPromise);
)

File: scripts/NativeCast.swift
import Foundation
import Capacitor

// ─── NativeCast iOS stub ──────────────────────────────────────────────────────
// Su iOS il casting nativo avviene tramite AirPlay (WebKit / AVRoutePickerView).
// Il Google Cast iOS SDK causa errori di modulo irrisolvibili con Xcode 16 +
// use_frameworks!, perciò questo plugin è un no-op che restituisce sempre
// { success: false, state: "unavailable" }.
// L'interfaccia JS (useChromecast.ts) vede stato "unavailable" su iOS nativo
// e non mostra il pulsante Chromecast: l'utente usa il pulsante AirPlay dedicato.
// ─────────────────────────────────────────────────────────────────────────────

@objc(NativeCastPlugin)
public class NativeCastPlugin: CAPPlugin, CAPBridgedPlugin {

    public let identifier    = "NativeCastPlugin"
    public let jsName        = "NativeCast"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "initialize",        returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "showPickerAndLoad", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "endSession",        returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getState",          returnType: CAPPluginReturnPromise),
    ]

    @objc func initialize(_ call: CAPPluginCall) {
        call.resolve(["success": false, "state": "unavailable"])
    }

    @objc func showPickerAndLoad(_ call: CAPPluginCall) {
        call.resolve(["success": false])
    }

    @objc func endSession(_ call: CAPPluginCall) {
        call.resolve(["success": false])
    }

    @objc func getState(_ call: CAPPluginCall) {
        call.resolve(["state": "unavailable"])
    }
}
